### ISSUE TITLE
Cursor no longer selects entire tab (accessibility issue)

### ISSUE THREAD

**Issue by dstpierre** [microsoft/vscode#159366]
Type: <b>Bug</b>

Placing the cursor over a tab character no longer selects the entire tab but only on "column".

For example, say I have tab size 4 and a solid block cursor the behavior before this version was:

[####] where # means the cursor block covered 4 columns.

[#___] now only the first column is selected/highlighted (the _ are to demonstrate spaces).

As a blind programmer not fully using a screen reader yet, this was very useful for me that the block cursor was selecting all "columns", it was very clear that the character was a tab.

Pinging  @isidorn as requested on gitter Microsoft/vscode-a11y

VS Code version: Code 1.70.2 (e4503b30fc78200f846c62cf8091b76ff5547662, 2022-08-16T05:36:25.715Z)
OS version: Linux x64 4.19.0-21-amd64
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770 CPU @ 3.40GHz (8 x 3654)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|0, 0, 0|
|Memory (System)|15.60GB (8.29GB free)|
|Process Argv|. --crash-reporter-id 9968ffc9-987e-4313-bff4-7ee39d8eff8b|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|undefined|
|XDG_CURRENT_DESKTOP|undefined|
|XDG_SESSION_DESKTOP|undefined|
|XDG_SESSION_TYPE|tty|
</details><details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
rails-db-schema|aki|0.2.5
elm-ls-vscode|elm|2.4.1
go|gol|0.35.2
vscode-test-explorer|hbe|2.21.1
elixir-ls|Jak|0.11.0
language-gettext|mro|0.2.0
csharp|ms-|1.25.0
python|ms-|2022.12.1
vscode-pylance|ms-|2022.8.40
jupyter|ms-|2022.7.1102252217
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.9
test-adapter-converter|ms-|0.1.6
phoenix|pho|0.1.1
ruby|reb|0.28.1
svelte-vscode|sve|105.21.0
vscode-ruby|win|0.28.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
wslgetstarted:30449410
cppdebug:30492333
pylanb8912:30545647
vsclangdf:30486550
c4g48928:30535728
hb751961:30553087
dsvsc012cf:30540253
azure-dev_surveyone:30548225
2144e591:30553903

```

</details>

**Comment by isidorn** [comment]
@dstpierre thanks a lot for filling this. And sorry for the slow response.
I can reproduce, steps:

1. Set `"editor.cursorStyle": "block"`
2. Navigate cursor to the tab character. Notice that the block cursor is only 1 character wide (it should be 4 characters wide for tab size 4)🐛 

I checked out previous VS Code versions and this was working fine in `1.69`. So this regressed with the `1.70` update

fyi @hediet for editor

**Comment by alexdima** [comment]
This was a change done for https://github.com/microsoft/vscode/issues/151959. I am sorry about the breakage here. @isidorn what do you recommend ?

**Comment by isidorn** [comment]
@alexdima I am not sure. I suggest we revert back to the old behaviour so Block cursor selects the entire tab.
Maybe we check what Notepad++ and SublimeText are doing.
What do you think?

**Comment by alexdima** [comment]
I've checked SublimeText and they render the block cursor with the width of 1ch when it is on a tab:

<img width="122" alt="image" src="https://user-images.githubusercontent.com/5047891/193227033-be59ec3e-b663-4a71-b648-ba166662ded0.png">

**Comment by isidorn** [comment]
Then how about we stick with the current behaviour and keep this issue open to collect feedback? Maybe too many users got accustomed to the previous "buggy" behaviour - but I am not convinced yet.

**Comment by dstpierre** [comment]
it's funny it's seen as a bug, I personally see this a a feature.

Could it be an option in config, say: `cursorSelectFullTab` (whatever the name) and this drive this behavior?

Or is this something I could implement via an extension (if I'm the only one finding this useful)?

**Comment by alexdima** [comment]
🤖 Thanks for raising this — I appreciate the accessibility perspective. The block cursor width on tabs was intentionally changed in https://github.com/microsoft/vscode/commit/a449148a14d to fix https://github.com/microsoft/vscode/issues/151959, and the current 1-character-width rendering matches other editors like SublimeText.

As discussed in the thread, the recommended path for restoring the old full-tab-width behavior is an extension using editor decorations. Closing as by-design, but please open a new issue if you'd like to request a built-in setting for this — that would be a feature request the team could evaluate.
